plugins/callback/logstash.py

```diff
@@ -63,76 +92,85 @@
     CALLBACK_TYPE = 'aggregate'
-    CALLBACK_NAME = 'community.general.logstash'
+    CALLBACK_NAME = 'logstash'
     CALLBACK_NEEDS_WHITELIST = True
 
-    def __init__(self, display=None):
-        super(CallbackModule, self).__init__(display=display)
+    def __init__(self):
+        super(CallbackModule, self).__init__()
 
         if not HAS_LOGSTASH:
             self.disabled = True
-            self._display.warning("The required python-logstash is not installed. "
-                                  "pip install python-logstash")
+            self._display.warning("The required python3-logstash is not installed. "
+                                  "pip install python3-logstash")
 
         self.start_time = datetime.utcnow()
 
-    def set_options(self, task_keys=None, var_options=None, direct=None):
+    def _init_plugin(self):
+        if not self.disabled:
+            self.logger = logging.getLogger('python-logstash-logger')
+            self.logger.setLevel(logging.DEBUG)
+
+            self.handler = logstash.LogstashHandler(
+                self.ls_server,
+                self.ls_port,
+                version=1,
+                message_type=self.ls_type
+            )
+
+            self.logger.addHandler(self.handler)
+            self.hostname = socket.gethostname()
+            self.session = str(uuid.uuid4())
+            self.errors = 0
+
+            self.base_data = {
+                'session': self.session,
+                'host': self.hostname
+            }
+
+            if self.ls_pre_command is not None:
+                self.base_data['ansible_pre_command_output'] = os.popen(
+                    self.ls_pre_command).read()
 
+            if self._options is not None:
+                self.base_data['ansible_checkmode'] = self._options.check
+                self.base_data['ansible_tags'] = self._options.tags
+                self.base_data['ansible_skip_tags'] = self._options.skip_tags
+                self.base_data['inventory'] = self._options.inventory
+
+    def set_options(self, task_keys=None, var_options=None, direct=None):
         super(CallbackModule, self).set_options(task_keys=task_keys, var_options=var_options, direct=direct)
 
-        self.logger = logging.getLogger('python-logstash-logger')
-        self.logger.setLevel(logging.DEBUG)
-
-        self.logstash_server = self.get_option('server')
-        self.logstash_port = self.get_option('port')
-        self.logstash_type = self.get_option('type')
-        self.handler = logstash.TCPLogstashHandler(
-            self.logstash_server,
-            int(self.logstash_port),
-            version=1,
-            message_type=self.logstash_type
-        )
-        self.logger.addHandler(self.handler)
-        self.hostname = socket.gethostname()
-        self.session = str(uuid.uuid1())
-        self.errors = 0
+        try:
+            self.ls_server = self.get_option('server')
+        except KeyError:
+            self.ls_server = "localhost"
```

Comment:
```suggestion
        self.ls_server = self.get_option('server')
```
Same for the others below.